### ISSUE TITLE
Training MobileNetV2 on Android device

### ISSUE THREAD

**Issue by MathieuMonney** [adap/flower#1454]
### Describe the bug

I'm trying to run the Android example from https://github.com/adap/flower/tree/main/examples/android with a custom model, in this case MobileNetV2.

I updated the head in the model conversion script (and batch size to train on one sample), swapped the model in the Android application with the converted MobileNetV2, and adapted the app so we don't need to have the backend running. I created a public repo [here](https://github.com/privately-corp/flower-android).
Inference seems to work fine, but training fails with the following stack trace:

```
flwr.android_client E/AndroidRuntime: FATAL EXCEPTION: Thread-2
    Process: flwr.android_client, PID: 15609
    java.lang.RuntimeException: Exception occurred during model training
        at flwr.android_client.TransferLearningModelWrapper.lambda$train$0$TransferLearningModelWrapper(TransferLearningModelWrapper.java:67)
        at flwr.android_client.-$$Lambda$TransferLearningModelWrapper$Rde6_7xRgPQh-TFtIUJ65ERGDRw.run(Unknown Source:4)
        at java.lang.Thread.run(Thread.java:1012)
     Caused by: java.lang.IllegalArgumentException: Model expected 261 parameter tensors, but got 260
        at org.tensorflow.lite.examples.transfer.api.LiteTrainHeadModel.calculateGradients(LiteTrainHeadModel.java:59)
        at org.tensorflow.lite.examples.transfer.api.TransferLearningModel.lambda$train$1$TransferLearningModel(TransferLearningModel.java:316)
        at org.tensorflow.lite.examples.transfer.api.-$$Lambda$TransferLearningModel$WVjn7-cmfDJ1v1c7WjX1Rt1Hh-g.call(Unknown Source:6)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
        at java.lang.Thread.run(Thread.java:1012) 
```

The error is thrown [here](https://github.com/privately-corp/flower-android/blob/6787f125af3009583e7a50adfd9ef9bc65fc61c8/client/transfer_api/src/main/java/org/tensorflow/lite/examples/transfer/api/LiteTrainHeadModel.java#L59). 

### Steps/Code to Reproduce

1. Clone the repo https://github.com/privately-corp/flower-android
2. Run the Android app under `client/` (needs NDK `20.1.5948944`)
3. In the app, load dataset partition 1
4. Click `Train one sample` to train the model on one sample, and crash the app

### Expected Results

The training step succeeds on the client.

### Actual Results

The app crashes with the following stack trace:

```
flwr.android_client E/AndroidRuntime: FATAL EXCEPTION: Thread-2
    Process: flwr.android_client, PID: 15609
    java.lang.RuntimeException: Exception occurred during model training
        at flwr.android_client.TransferLearningModelWrapper.lambda$train$0$TransferLearningModelWrapper(TransferLearningModelWrapper.java:67)
        at flwr.android_client.-$$Lambda$TransferLearningModelWrapper$Rde6_7xRgPQh-TFtIUJ65ERGDRw.run(Unknown Source:4)
        at java.lang.Thread.run(Thread.java:1012)
     Caused by: java.lang.IllegalArgumentException: Model expected 261 parameter tensors, but got 260
        at org.tensorflow.lite.examples.transfer.api.LiteTrainHeadModel.calculateGradients(LiteTrainHeadModel.java:59)
        at org.tensorflow.lite.examples.transfer.api.TransferLearningModel.lambda$train$1$TransferLearningModel(TransferLearningModel.java:316)
        at org.tensorflow.lite.examples.transfer.api.-$$Lambda$TransferLearningModel$WVjn7-cmfDJ1v1c7WjX1Rt1Hh-g.call(Unknown Source:6)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
        at java.lang.Thread.run(Thread.java:1012) 
```

Note that the number of parameters in the error might be misleading, the code throwing the error is

```
if (modelWrapper.getInterpreter().getOutputTensorCount() != modelParameters.length + 1) {
  throw new IllegalArgumentException(String.format(
      "Model expected %d parameter tensors, but got %d",
      modelWrapper.getInterpreter().getInputTensorCount() - 1,
      modelParameters.length));
}
```
Which I think should print `getOutputTensorCount` instead of `getInputTensorCount`. In that case the error is `Model expected 156 parameter tensors, but got 260`.

**Comment by MathieuMonney** [comment]
Turns out some layers in MobileNetV2 are not supported when training (BatchNorm is one of them). Using MobileNetV2 as base layer and adding a training head works fine.
